add doc for new-collation-enable in br (#8809)
pingcap/docs-cn: br/backup-and-restore-tool.md

| 用 BR v4.0 备份 TiDB v4.0 | ✅ | ✅  | ✅（如果 TiKV >= v4.0.0-rc.1，BR 包含 [#233](https://github.com/pingcap/br/pull/233) Bug 修复，且 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241) Bug 修复，那么 BR 会导致 TiKV 节点重启) |
| 用 BR nightly 或 v5.0 备份 TiDB v4.0 | ❌（当 TiDB 版本小于 v4.0.9 时会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |

> #### new_collations_enabled_on_first_bootstrap 检查

Reviewer: 最好检查下 TiDB 相关文档是否有类似描述，可以做个关联

> #### new_collations_enabled_on_first_bootstrap 检查

Reviewer 2: ```suggestion
#### 对 `new_collations_enabled_on_first_bootstrap` 进行检查
```

> 从 TiDB v6.0 版本开始，[`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置的默认值由 `false` 改为 `true`，当上下游集群的此项配置相同时，才可以将上游集群的备份数据安全地恢复到下游集群中，具体原因见[排序规则支持](/character-set-and-collation.md#排序规则支持)。

Reviewer 2: [排序规则支持](https://github.com/joccau/docs-cn/blob/fix-check-collation/character-set-and-collation.md#%E6%8E%92%E5%BA%8F%E8%A7%84%E5%88%99%E6%94%AF%E6%8C%81) 里有提到 BR 不能在上下游该配置项不同时进行数据恢复的原因吗？
```suggestion
从 TiDB v6.0.0 版本开始，[`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置项的默认值由 `false` 改为 `true`。当上下游集群的此项配置相同时，BR 才会将上游集群的备份数据安全地恢复到下游集群中。
```

> 从 br v6.0 开始，br 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。如果用户需要将旧版本的备份数据恢复到 TiDB v6.0 中，用户需要自行检查上下游集群中的此配置项是否相同，若相同，则在恢复命令中添加 `--check-requirements=false` 可跳过此项配置检查。

Reviewer 2: 如果上下游的 `new_collations_enabled_on_first_bootstrap` 配置项不相同会怎样呢？会有什么报错呢？

> 从 br v6.0 开始，br 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。若配置不相同，会拒绝恢复，并报告此配置项不匹配的错误。如果用户需要将旧版本的备份数据恢复到 TiDB v6.0 中，用户需要自行检查上下游集群中的此配置项是否相同，若相同，则在恢复命令中添加 `--check-requirements=false` 可跳过此项配置检查，若配置不相同的情况下强行恢复，会报告[数据校验错误](/br/backup-and-restore-tool.md#使用限制)。

Reviewer 2: ```suggestion
从 v6.0.0 开始，BR 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。若上下游的该配置不相同，BR 会拒绝恢复，并报告此配置项不匹配的错误。

如果你需要将旧版本的备份数据恢复到 TiDB v6.0.0 或更新版本的 TiDB 集群中，你需要自行检查上下游集群中的该配置项是否相同：

- 若该配置该相同，则可在恢复命令中添加 `--check-requirements=false` 以跳过此项配置检查。
- 若该配置不相同，且进行强行恢复，BR 会报告[数据校验错误](/br/backup-and-restore-tool.md#使用限制)。
```
